src/java/org/apache/cassandra/triggers/TriggerExecutor.java

```diff
@@ -252,8 +272,23 @@ private List<Mutation> executeInternal(PartitionUpdate update)
         }
     }
 
+    public synchronized void loadTriggerClass(String triggerClass) throws Exception
+    {
+        // Allow loading the class regardless of Config, since this could happen as part of TCM replay via
+        // CreateTriggerStatement#apply.
+        // Check that triggerClass is available on the classpath, but do not initialize the class since that would
+        // execute static blocks.
+        customClassLoader.loadClass(triggerClass).getConstructor();
+    }
+
     public synchronized ITrigger loadTriggerInstance(String triggerClass) throws Exception
     {
+        Config.TriggersPolicy policy = DatabaseDescriptor.getTriggersPolicy();
+        if (policy == Config.TriggersPolicy.disabled || policy == Config.TriggersPolicy.forbidden)
```

Comment:
@aratno can't be this just simplitied to `if (policy != enabled)` ?

Reply:
I could, it's a personal style preference to have enum checks be exhaustive / explicit. I find that it makes usage of each enum value easier to understand. I'd do `!=` if there were so many fields it felt wrong to list.